src/test/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStepTest.java

```diff
@@ -186,16 +197,26 @@ public InvalidChangelogSCM() { }
         @Override public void checkout(Run<?,?> build, Launcher launcher, FilePath workspace, TaskListener listener, File changelogFile, SCMRevisionState baseline) throws IOException, InterruptedException {
             // Unlike the superclass, which adds an XML header to the file, we just ignore the file.
         }
-        @Override public ChangeLogParser createChangeLogParser() {
-            return new ChangeLogParser() {
-                public ChangeLogSet<? extends ChangeLogSet.Entry> parse(Run build, RepositoryBrowser<?> browser, File changelogFile) throws IOException, SAXException {
-                    Jenkins.XSTREAM2.fromXML(changelogFile);
-                    throw new AssertionError("The previous line should always fail because the file is empty");
-                }
-            };
-        }
+        ChangeLogParser changeLogParser = createChangeLogParser();
         @TestExtension("scmParsesUnmodifiedChangelogFile")
         public static class DescriptorImpl extends NullSCM.DescriptorImpl { }
     }
 
+    public static class FakeChangeLogSCM extends NullSCM {
+        @DataBoundConstructor
+        public FakeChangeLogSCM() { }
+        ChangeLogParser changeLogParser = createChangeLogParser();
```

Comment:
But I don't think this is not calling the method you think it is.  It is not calling `createChangeLogParser()` on line 213.  It is  actually calling `super.createChangeLogParser()` - the instance method defined in `NullSCM` or above.